InvaderZim85/MsSqlToolBelt
# Class Gen settings, Replcation filter fix

## Whats changed?

- Added the following options for the class generator:
   - *Save class generator options*: If selected, the selected options of the class generator (*Add summary, DB Model (EF), etc.) will be saved when you create a class. The next time the program is started, the options from the last time are loaded and set accordingly (default = `false`)
   - *Default class modifier*: Here you can set the *default* modifier which should be used every time (default = `public`).
   - *Hide SQL Query*: The SQL query with which the data can be loaded is displayed below the generated class. However, if you do not need this information, you can select this option and the display of the SQL query will be hidden.
- Fixed a bug in the *Replication info*: The filter was not reset correctly when connecting to another server / database
- Update of the nuget packages

## Screenshots

![image](https://github.com/user-attachments/assets/0f4e78ef-f80e-4a00-9cb4-0b64bf0ece36)

## How to test?

1. Open the program
2. Open the settings (icon on the top right)
3. Change the settings
4. Open the class generator and set the options
5. Generate a class
6. Close the program
7. Start the program
8. Depending on the settings, the modifier options are set accordingly. Furthermore, the SQL query info is displayed or not.

## Checklist

Complete the following checklist to make sure you haven't forgotten anything

- [x] Unnecessary comments have been removed
- [x] TODOs have been edited or provided with appropriate information
- [x] Usings have been cleaned up
- [x] All required files have been committed
- [x] Code has been reviewed by the developer
